javascript/javascript3/promisExercises.js

```diff
@@ -0,0 +1,75 @@
+// this is happening - backend, server, API
+// function YesNoFail4Seconds() {
+//   console.log("Start getting yes/no");
+//   return new Promise((resolve, reject) => {
+//     const decision = Math.floor(Math.random() * 3);
+//     console.log(`decision = ${decision}`);
+//     setTimeout(() => {
+//       if (decision === 0) resolve("yes");
+//       else if (decision === 1) resolve("no");
+//       else reject("failure!!!");
+//     }, 4000);
+//   });
+// }
+// //you make this code (frontend)
+// YesNoFail4Seconds()
+//   .then((returnedHelloObject) => {
+//     console.log(`The answer is ${returnedHelloObject}`);
+//   })
+//   .catch((error) => {
+//     console.log(error);
+//   });
+  
+/////exercise 4
+// getLoggedInUsers should wait 5 seconds before it resolves with the users! 
+// Sometimes it should resolve and sometimes it should reject. 
+// Look into Math.random
+// getLoggedInUsers()
+// .then(users => {
+//     console.log(users); // ['benna', 'magdy', 'carolina']
+// }).catch(error => {
+//     console.log(error);
+// });
+
+   async function getLoggedInUsers() {
+          console.log("Start getting yes/no");
+          return new Promise((resolve, reject) => {
+            const users = ['benna', 'magdy', 'carolina']; 
+            const result = Math.floor(Math.random() * 5);
+            console.log(result);
+            setTimeout(() => {
+              if (users === 0) resolve("yes");
+              else if (users === 1) resolve("no");
```

Comment:
This condition will always return 'false' since the types 'string[]' and 'number' have no overlap.